bound/kt/src/test/kotlin/web5/sdk/vc/VerifiableCredentialTest.kt

```diff
@@ -0,0 +1,286 @@
+        assertEquals(issuer, vc.issuer)
+    }
+
+    @Test
+    fun test_issuer_object_supports_additional_properties() {
+        this.testSuite.include()
+        val additionalProperties = mapOf("extra_key" to "extra_value")
+
+        val issuer = Issuer.ObjectIssuer(
+            ISSUER_DID_URI,
+            "Example Name",
+            additionalProperties
+        )
+
+        val vc = VerifiableCredential.create(issuer, CREDENTIAL_SUBJECT, VerifiableCredentialCreateOptions())
+
+        if (vc.issuer is Issuer.ObjectIssuer) {
+            assertEquals(additionalProperties, (vc.issuer as Issuer.ObjectIssuer).additionalProperties)
+        } else {
+            fail("Issuer is not an ObjectIssuer")
+        }
+    }
+
+    @Test
+    fun test_credential_subject_id_must_not_be_empty() {
+        this.testSuite.include()
+        val credentialSubject = CredentialSubject("")
+
+        try {
+            VerifiableCredential.create(ISSUER, credentialSubject, VerifiableCredentialCreateOptions())
+            fail("Expected an exception to be thrown")
+        } catch (e: Web5Exception) {
+            when (e) {
+                is Web5Exception.Exception -> {
+                    assertEquals(e.msg, "parameter error subject id must not be empty")
+                }
+                else -> {
+                    fail("Caught an unexpected exception type")
+                }
+            }
+        }
+    }
+
+    @Test
+    fun test_credential_subject_must_be_set() {
+        this.testSuite.include()
+        val vc = VerifiableCredential.create(ISSUER, CREDENTIAL_SUBJECT, VerifiableCredentialCreateOptions())
+        assertEquals(CREDENTIAL_SUBJECT, vc.credentialSubject)
+    }
+
+    @Test
+    fun test_credential_subject_supports_additional_properties() {
+        this.testSuite.include()
+        val additionalProperties = mapOf("extra_key" to "extra_value")
+
+        val credentialSubject = CredentialSubject(
+            SUBJECT_DID_URI,
+            additionalProperties
+        )
+
+        val vc = VerifiableCredential.create(ISSUER, credentialSubject, VerifiableCredentialCreateOptions())
+        assertEquals(additionalProperties, vc.credentialSubject.additionalProperties)
+    }
+
+    @Test
+    fun test_issuance_date_must_be_set() {
+        this.testSuite.include()
+        val issuanceDate = Date()
+
+        val options = VerifiableCredentialCreateOptions(
+            issuanceDate = issuanceDate
+        )
+
+        val vc = VerifiableCredential.create(ISSUER, CREDENTIAL_SUBJECT, options)
+        assertEquals(issuanceDate, vc.issuanceDate)
+    }
+
+    @Test
+    fun test_issuance_date_must_be_now_if_not_supplied() {
+        this.testSuite.include()
+        val vc = VerifiableCredential.create(ISSUER, CREDENTIAL_SUBJECT, VerifiableCredentialCreateOptions())
+
+        val now = Date()
+        val oneSecondAgo = Date(now.time - 1000)
```

Comment:
maybe set to 1 year or something incase this github action nondeterministically lags at the wrong time or something lol

Reply:
meh, we'd just re-run the test

maybe I could make it 10 seconds tho, that seems reasonable to me... done!